CMakeLists.txt

```diff
@@ -358,24 +358,60 @@ ENDIF()
 
 # Optional external dependency: libssh2
 IF (USE_SSH)
-	PKG_CHECK_MODULES(LIBSSH2 libssh2)
-ENDIF()
-IF (LIBSSH2_FOUND)
-	ADD_DEFINITIONS(-DGIT_SSH)
-	INCLUDE_DIRECTORIES(${LIBSSH2_INCLUDE_DIRS})
-	LINK_DIRECTORIES(${LIBSSH2_LIBRARY_DIRS})
-	LIST(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
-	#SET(LIBGIT2_PC_LIBS "${LIBGIT2_PC_LIBS} ${LIBSSH2_LDFLAGS}")
-	SET(SSH_LIBRARIES ${LIBSSH2_LIBRARIES})
-
-	CHECK_LIBRARY_EXISTS("${LIBSSH2_LIBRARIES}" libssh2_userauth_publickey_frommemory "${LIBSSH2_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
-	IF (HAVE_LIBSSH2_MEMORY_CREDENTIALS)
-		ADD_DEFINITIONS(-DGIT_SSH_MEMORY_CREDENTIALS)
+	IF (PKG_CONFIG_EXECUTABLE_FOUND AND NOT WIN32)
+		PKG_CHECK_MODULES(LIBSSH2 libssh2)
+		
+		IF (LIBSSH2_FOUND)
+			ADD_DEFINITIONS(-DGIT_SSH)
+			INCLUDE_DIRECTORIES(${LIBSSH2_INCLUDE_DIRS})
+			LINK_DIRECTORIES(${LIBSSH2_LIBRARY_DIRS})
+			LIST(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
+			#SET(LIBGIT2_PC_LIBS "${LIBGIT2_PC_LIBS} ${LIBSSH2_LDFLAGS}")
+			SET(SSH_LIBRARIES ${LIBSSH2_LIBRARIES})
+
+			CHECK_LIBRARY_EXISTS("${LIBSSH2_LIBRARIES}" libssh2_userauth_publickey_frommemory "${LIBSSH2_LIBRARY_DIRS}" HAVE_LIBSSH2_MEMORY_CREDENTIALS)
+			IF (HAVE_LIBSSH2_MEMORY_CREDENTIALS)
+				ADD_DEFINITIONS(-DGIT_SSH_MEMORY_CREDENTIALS)
+			ENDIF()
+		ELSE()
+			MESSAGE(STATUS "LIBSSH2 not found. Set CMAKE_PREFIX_PATH if it is installed outside of the default search path.")
+		ENDIF()
+	ELSE()
+		# pkg-config is not the ideal way to find dependencies on Windows. Use the standard
+		# CMake way of find_package on Windows, or if pkg-config is not available on
+		# non-Windows operating systems.
+		FIND_PACKAGE(libssh2)
+		
+		IF (LIBSSH2_FOUND)
+			ADD_DEFINITIONS(-DGIT_SSH)
+			
+			# TODO: Determine the contents of LIBSSH2_LDFLAGS.
+			#LIST(APPEND LIBGIT2_PC_LIBS ${LIBSSH2_LDFLAGS})
```

Comment:
If I'm not mistaken, the target returned by `find_package` uses absolute paths for the library. If so, there should be no need for setting LDFLAGS, but I wasn't able to find it in CMake's documentation.